experimenter/experimenter/jetstream/models.py

```diff
@@ -159,34 +159,39 @@ def append_retention_data(self, weekly_data):
 
 
 class DataPoint(BaseModel):
-    lower: float = None
-    upper: float = None
-    point: float = None
-    window_index: str = None
-    count: float = None
+    lower: float | None = None
+    upper: float | None = None
+    point: float | None = None
+    window_index: str | None = None
+    count: float | None = None
 
-    def set_window_index(self, window_index):
-        self.window_index = window_index
+    @field_validator("window_index", mode="before")
+    @classmethod
+    def coerce_window_index(cls, v: Any):
+        if isinstance(v, int):
+            return str(v)
+
+        return v
 
     def has_bounds(self) -> bool:
-        return self.lower and self.upper
+        return self.lower and self.upper # TODO: should these be `is not None` checks?
```

Comment:
So thinking about this a bit more.. I'd say yes we should change to `is not None` checks.

Looks like the current evaluation is wrong if `lower` or `upper` is 0, which would be falsey and so the function would erroneously return `False`. Because we have default values of `None` for these fields, the value should either be explicitly `None` or a valid `float`, so checking `is not None` is correct here.